Codegen/src/Nodes/TypedDataNode.cs

```diff
@@ -486,9 +486,11 @@ internal static List<ITextLine> LineToTextLines(string line)
             }
 
             list_items.AddRange(pathVariables);
-            string qn = string.Join(".", list_items.ToArray());
-            AddIndexMap(rootNode, rootNode.QualifiedName.ToString(), indexes, map);
-            AddIndexMap(ownerDefinition, qn, indexes, map);
+            string qn = string.Join(".", list_items.Where(item => item != null));
+
+            // If indexes directly come from their DataDef, use the computed path (qn). If they come from a TypeDef, the path is the rootNode.QualifiedName.
+            AddIndexMap(rootNode, rootNode == ownerDefinition ? qn : rootNode.QualifiedName.ToString(), indexes, map);
```

Comment:
For visibility reason the line 492 should be
AddIndexMap(rootNode, (rootNode == ownerDefinition ? qn : rootNode.QualifiedName.ToString()), indexes, map);